unstructured/file_utils/filetype.py

```diff
@@ -106,100 +112,217 @@ def __lt__(self, other):
 }
 
 
+def _detect_filetype_from_extension(extension: str) -> Optional[FileType]:
+    """Detect the file type based on the file extension."""
+    if extension in [".txt", ".text"]:
+        return FileType.TEXT
+    elif extension in [".pdf"]:
+        return FileType.PDF
+    elif extension in [".jpg", ".jpeg", ".png", ".gif"]:
+        return FileType.IMAGE
+    # Add more supported file types and their extensions here.
+    return None
+
+
+def _detect_filetype_from_extension(extension: str) -> Optional[FileType]:
+    """Detect the file type based on the file extension."""
+    if extension in [".txt", ".text"]:
+        return FileType.TEXT
+    elif extension in [".pdf"]:
+        return FileType.PDF
+    elif extension in [".jpg", ".jpeg", ".png", ".gif"]:
+        return FileType.IMAGE
+    # Add more supported file types and their extensions here.
+    return None
+
+
 def detect_filetype(
     filename: Optional[str] = None, file: Optional[IO] = None
 ) -> Optional[FileType]:
-    """Use libmagic to determine a file's type. Helps determine which partition brick
-    to use for a given file. A return value of None indicates a non-supported file type."""
-    if filename and file:
-        raise ValueError("Only one of filename or file should be specified.")
-
-    if filename:
-        _, extension = os.path.splitext(filename)
-        extension = extension.lower()
-        mime_type = magic.from_file(filename, mime=True)
-    elif file is not None:
-        extension = None
-        # NOTE(robinson) - the python-magic docs recommend reading at least the first 2048 bytes
-        # Increased to 4096 because otherwise .xlsx files get detected as a zip file
-        # ref: https://github.com/ahupp/python-magic#usage
-        mime_type = magic.from_buffer(file.read(4096), mime=True)
-    else:
-        raise ValueError("No filename nor file were specified.")
+    try:
+        # Attempt to load the libmagic library
+        magic_obj = magic.Magic()
+        mime_type = magic_obj.from_file(filename, mime=True)
+        """Use libmagic to determine a file's type. Helps determine which partition brick
+        to use for a given file. A return value of None indicates a non-supported file type."""
+        if filename and file:
+            raise ValueError("Only one of filename or file should be specified.")
+
+        if filename:
+            _, extension = os.path.splitext(filename)
+            extension = extension.lower()
+            mime_type = magic.from_file(filename, mime=True)
+            if not LIBMAGIC_AVAILABLE:
```

Comment:
Move this check above `mime_type = magic.from_file(filename, mime=True)`, otherwise you'll get an error if `libmagic` isn't installed